node/pkg/watchers/algorand/watcher.go

```diff
@@ -167,135 +156,110 @@ func (e *Watcher) Run(ctx context.Context) error {
 
-	go func() {
-		timer := time.NewTicker(time.Second * 1)
-		defer timer.Stop()
+	timer := time.NewTicker(time.Second * 1)
+	defer timer.Stop()
 
-		indexerClient, err := indexer.MakeClient(e.indexerRPC, e.indexerToken)
-		if err != nil {
-			logger.Error("indexer make client", zap.Error(err))
-			p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
-			errC <- err
-			return
-		}
+	indexerClient, err := indexer.MakeClient(e.indexerRPC, e.indexerToken)
+	if err != nil {
+		logger.Error("indexer make client", zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+		return err
+	}
 
-		algodClient, err := algod.MakeClient(e.algodRPC, e.algodToken)
-		if err != nil {
-			logger.Error("algod client", zap.Error(err))
-			p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
-			errC <- err
-			return
-		}
+	algodClient, err := algod.MakeClient(e.algodRPC, e.algodToken)
+	if err != nil {
+		logger.Error("algod client", zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+		return err
+	}
 
-		if e.next_round == 0 {
-			status, err := algodClient.StatusAfterBlock(0).Do(context.Background())
-			if err != nil {
-				logger.Error("StatusAfterBlock", zap.Error(err))
-				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
-				errC <- err
-				return
-			}
+	status, err := algodClient.StatusAfterBlock(0).Do(context.Background())
+	if err != nil {
+		logger.Error("StatusAfterBlock", zap.Error(err))
+		p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+		return err
+	}
 
-			e.next_round = status.NextVersionRound
-		}
+	e.next_round = status.NextVersionRound
 
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case r := <-e.obsvReqC:
-				if vaa.ChainID(r.ChainId) != vaa.ChainIDAlgorand {
-					panic("invalid chain ID")
-				}
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case r := <-e.obsvReqC:
+			if vaa.ChainID(r.ChainId) != vaa.ChainIDAlgorand {
+				panic("invalid chain ID")
+			}
 
-				logger.Info("Received obsv request",
-					zap.String("tx_hash", hex.EncodeToString(r.TxHash)),
-					zap.String("base32_tx_hash", base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash)))
+			logger.Info("Received obsv request",
+				zap.String("tx_hash", hex.EncodeToString(r.TxHash)),
+				zap.String("base32_tx_hash", base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash)))
 
-				result, err := indexerClient.SearchForTransactions().TXID(base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash)).Do(context.Background())
+			result, err := indexerClient.SearchForTransactions().TXID(base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash)).Do(context.Background())
+			if err != nil {
+				logger.Error("SearchForTransactions", zap.Error(err))
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+				break
+			}
+			for i := 0; i < len(result.Transactions); i++ {
```

Comment:
can do `for _, t := range result.Transactions`

Reply:
learning go.. :)